plugins/scaffolder-backend/src/scaffolder/actions/builtin/bitbucket-cloud/helpers.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch, { Response, RequestInit } from 'node-fetch';
+
+/**
+ * Uses the Bitbucket Cloud REST API to create a new repository.
+ * @public
+ */
+export const bitbucketCloudCreateRepository = async (opts: {
+  workspace: string;
+  project: string;
+  repo: string;
+  description?: string;
+  repoVisibility: 'private' | 'public';
+  mainBranch: string;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const {
+    workspace,
+    project,
+    repo,
+    description,
+    repoVisibility,
+    mainBranch,
+    authorization,
+    apiBaseUrl,
+  } = opts;
+
+  const options: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      scm: 'git',
+      description: description,
+      is_private: repoVisibility === 'private',
+      project: { key: project },
+    }),
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  let response: Response;
+  try {
+    response = await fetch(
```

Comment:
I would strongly suggest to use `@backstage/plugin-bitbucket-cloud-common` for communicating with the Bitbucket Cloud API.

Reply:
and related to that:
it may make sense to move this into a separate module (similar for other) like it was done at the catalog-backend, too.

Reply:
Thank you for this suggestion! I will look into using the bitbucket-cloud-common plugin. I knew I was missing something when I had to write the response schema while writing tests. As for the larger discussion of separating this into another module, I will reply below.